From 85d88e4d6b26eed925cbe62f0a56c1ec9e92033a Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 22 Apr 2022 17:31:19 +0200
Subject: [PATCH 111/N] configure.ac: Default to --without-c-locale-coercion
 on Windows

--with-c-locale-coercion otherwise defaults to yes and enables code
that isn't compatible on Windows, mainly because the feature is Unix related.

Default to "no" on Windows instead.

Fixes #36
---
 configure.ac | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4b77fed..3bc089b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4690,11 +4690,14 @@ AC_MSG_RESULT($with_freelists)
 AC_MSG_CHECKING(for --with-c-locale-coercion)
 AC_ARG_WITH(c-locale-coercion,
             AS_HELP_STRING([--with-c-locale-coercion],
-              [enable C locale coercion to a UTF-8 based locale (default is yes)]))
+              [enable C locale coercion to a UTF-8 based locale (default is yes on Unix, no on Windows)]))
 
 if test -z "$with_c_locale_coercion"
 then
-    with_c_locale_coercion="yes"
+    case $host in
+      *-*-mingw*) with_c_locale_coercion="no";;
+      *) with_c_locale_coercion="yes";;
+    esac
 fi
 if test "$with_c_locale_coercion" != "no"
 then
